maraboupy/test/test_nnet_extensions.py

```diff
@@ -0,0 +1,149 @@
+
+from maraboupy.MarabouNetworkNNet import *
+
+
+from subprocess import call
+
+property_filename = "./resources/properties/acas_property_4.txt"
+network_filename = "./resources/nnet/acasxu/ACASXU_experimental_v2a_1_9.nnet"
+
+layer = 2
+
+
+def test_write_to_file():
+    output_filename = "./maraboupy/test/ACASXU_experimental_v2a_1_9_output.nnet"
+
+    nnet_object = MarabouNetworkNNet(filename=network_filename)
+    nnet_object.writeNNet(output_filename)
+
+    call(['diff', output_filename,network_filename])
+
+
+def test_evaluate_network():
+    nnet_object = MarabouNetworkNNet(filename=network_filename)
+
+    N = 10
+    for i in range(N):
+        inputs = nnet_object.createRandomInputsForNetwork()
+
+        output1 = nnet_object.evaluateNetworkToLayer(inputs, last_layer=-1, normalize_inputs=False,
+                                                     normalize_outputs=False, activate_output_layer=False)
+
+        output2 = nnet_object.evaluateNetwork(inputs,normalize_inputs=False,normalize_outputs=False)
+
+        assert (output1 == output2).all()
+
+        # Compare evaluation with and without Marabou
+        without_marabou_output = nnet_object.evaluate(np.array([inputs]),useMarabou=False).flatten()
+        without_marabou_output_rounded = np.array([float(round(y,8)) for y in without_marabou_output])
+
+        with_marabou_output = nnet_object.evaluate(np.array([inputs]),useMarabou=True).flatten()
+        with_marabou_output_rounded = np.array([float(round(y,8)) for y in with_marabou_output])
+
+        # Assert that all of the above agree, at least up to 10^-8
+        assert (output1 == without_marabou_output).all()
```

Comment:
Rather than rounding and asserting equality, the absolution value of the difference can be asserted to be less than some tolerance value (such as 1e-8). Making tolerance a global variable allows the value to be easily changed for all asserts. See the other test files as an example.

Reply:
Done.